File: src/source-path.ts
export const DEPRECATED_SYN_EXTENSION_WARNING =
  ".syn files are deprecated. Use .sd files instead.";

/** Warn when a path uses the legacy `.syn` extension (still accepted). */
export function warnDeprecatedSourceExtension(filePath: string): void {
  if (filePath.endsWith(".syn")) {
    console.warn(DEPRECATED_SYN_EXTENSION_WARNING);
  }
}

File: crates/spanda-core/src/source_path.rs
pub const DEPRECATED_SYN_EXTENSION_WARNING: &str =
    ".syn files are deprecated. Use .sd files instead.";

/// Emit a deprecation warning when the path uses the legacy `.syn` extension.
pub fn warn_deprecated_source_extension(path: &str) {
    if path.ends_with(".syn") {
        eprintln!("warning: {DEPRECATED_SYN_EXTENSION_WARNING}");
    }
}
